src/views/UrbitOS/Home.tsx

```diff
@@ -42,22 +43,40 @@ export default function UrbitOSHome() {
     names,
   ]);
 
-  const { code, download } = useSingleKeyfileGenerator({});
+  const { available, code, download, generating } = useSingleKeyfileGenerator(
+    {}
+  );
+  const keyfileAvailable = available && !generating;
+
+  const headerButton = () => {
+    if (!hasSetNetworkKeys) {
+      return null;
+    }
+
+    if (generating) {
+      return <Box className="header-button keyfile">Generating...</Box>;
+    }
+
+    return keyfileAvailable ? (
+      <Button className="header-button keyfile" onClick={download}>
+        <KeyfileIcon />
+        Download Keyfile
+      </Button>
+    ) : (
+      <WithTooltip content="Either the key was derived non-deterministically, or try Login > Use Legacy Compatibility">
```

Comment:
This tooltip is super awkward lol.

Here and everywhere in the codebase, I would probably not use "legacy". Implies it was ever intentional, but it was really just a temporary buggy state. Probably "bugcompat" or w/e for variable names, not sure how to succinctly communicate this on the frontend.

...On further thinking, maybe the user-facing side is the only place "legacy" makes sense to use?

What you really want to communicate is that "if you generated a keyfile between 2021-05 and 2022-02, and you expected it here, try the login thing instead". But that feels _way_ too verbose for a tooltip...

We also probably want to put a little (i) here to indicate there is a tooltip in the first place.

Reply:
Yeah I think a an (i) icon inside the button would help here. 

I also think the wording could be different, trying to come up with something. Could we possibly just do the relogin for them on a button press? It almost seems like this is worthy of a modal just so we could fit a bit more text and maybe an action.

Just spitballing here, some bits may be wrong:
"We're unable to derive your keyfile. This is likely because it was generated using a signing method from the last six months or derived non-deterministically. Try logging in again with the previous signing method." and then have a button that does the relogin or brings them to the login form. I'm somewhat worried they won't find it behind the gear so we may need extra language to help them find it if we just redirect to the login page.

Reply:
@Fang- Thanks for the feedback. Cleaned up the UX based on this comment, but still not happy with the naming. Will think about how we can better label things more clearly for users.

Reply:
An update on this, I have looped in @jyng so we can leverage his expertise.